src/show.jl

```diff
@@ -22,9 +22,21 @@ Base.show(io::IO, ::ViewAxis{Inds, IdxMap, <:Ax}) where {Inds, IdxMap, Ax} =
 Base.show(io::IO, ::ViewAxis{Inds, IdxMap, <:NullorFlatAxis}) where {Inds, IdxMap} = 
     print(io, Inds)
 
+# Show reducted Axis types to avoid excessive stacktraces
+Base.show(io::IO, ::Type{Axis{IdxMap}}) where {IdxMap} = print(io, "Axis{...}")
+Base.show(io::IO, ::Type{FlatAxis}) = print(io, "FlatAxis")
+Base.show(io::IO, ::Type{NullAxis}) = print(io, "NullAxis")
+function Base.show(io::IO, ::Type{PartitionedAxis{PartSz,IdxMap,Ax}}) where {PartSz,IdxMap,Ax}
+    return print(io, "PartitionedAxis{$PartSz, {...}, $Ax}")
+end
+Base.show(io::IO, ::Type{ShapedAxis{Shape,IdxMap}}) where {Shape,IdxMap} = print(io, "ShapedAxis($Shape, {...})")
+Base.show(io::IO, ::Type{ViewAxis{Inds,IdxMap,Ax}}) where {Inds,IdxMap,Ax} = print(io, "ViewAxis{$Inds, {...}, $Ax)")
```

Comment:
This is how ComponentArrays used to do type printing, but we had to get rid of it because it [tends to break a lot of things](https://github.com/JuliaLang/julia/issues/22363)

Reply:
`Base.print_type_stacktrace` was supposed to be the solution to this, but if it’s not working anymore, I’m not sure what to do. We don’t want to potentially cause a segfault for someone.

Reply:
Ah, I should have thought that I am reinventing the wheel here 😄 

I took a look at the current `Julia`'s `show` code that is used for stacktraces but I found it too complicated, and it made me think that overriding something "private" there might be dangerous. 

How about:
* We open a new discourse thread to discuss this again with the community, since no good solution seems to exist as of now
* Keep overriding the `show` as per this PR, but only when a [preference is set](https://github.com/JuliaPackaging/Preferences.jl). Update the Readme accordingly.